### PR TITLE
Upgrade CircleCI to version 2

### PR DESCRIPTION
**Pull request by alaa-awartany**

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -1,40 +0,0 @@
-machine:
-  node:
-    version: 4.4.0
-
-general:
-  build_dir: .
-  artifacts:
-    - "css" # Compiled adk.css file
```
Comment (alaa-awartany) [inline circle.yml]: @awartani do you know how to migrate this section to v2, is this correct:
```
    working_directory: .
    ...
      - store_artifacts:
        path: .
        destination: css
```

File: circle.yml [hunk 2]
```diff
@@ -1,40 +0,0 @@
...
-    owner: Architizer
-    branch: /.*?/
-    commands:
-      - bash -x deploy.sh
-
-experimental:
-  notify:
-    branches:
-      only:
-        - master
```
Comment (alaa-awartany) [inline circle.yml]: @awartani any idea what this section does? how to migrate it to v2?

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      - run: npm install -g gulp
+
+      - run:
+        name: compile sass and create docs
+        command: |
+          npm install
+          gulp sass
+          gulp docs
+          gulp createAdkTemplatesList
```
Comment (alaa-awartany) [inline .circleci/config.yml]: @awartani CircleCI is failing on this (where ever there is a multiline command) with the following error `has incorrect indentation, it should be formatted like:`.
https://circleci.com/gh/Architizer/design-kit/923

any idea why it's not seeing the indentation of multi-lined commands?

Comment (awartani) [reply .circleci/config.yml]: @alaa-awartany you have 10 spaces, and it should be 12. 
```
            npm install
            gulp sass
            gulp createAdkTemplatesList
```

Don't trust the editor to have the right spaces for you

File: circle.yml (hunk 1 above)
Comment (awartani) [reply circle.yml]: why do you need `artifacts`?
you find a lot of documentation on CircleCI about this, but you can find an example I wrote for mobile app https://github.com/Architizer/architizer-mobile-app/blob/master/.circleci/config.yml#L90

File: circle.yml (hunk 2 above)
Comment (awartani) [reply circle.yml]: No idea, first time seeing this.

File: circle.yml (hunk 1 above)
Comment (alaa-awartany) [reply circle.yml]: it's being used to store the compiled css file which gets moved to S3.
https://github.com/Architizer/design-kit/blob/master/deploy.sh#L8

File: .circleci/config.yml (hunk 3 above)
Comment (alaa-awartany) [reply .circleci/config.yml]: when i increased them to 12, i got `Invalid step structure`.
https://circleci.com/gh/Architizer/design-kit/926#tests/containers/0

**Comment by alaa-awartany** [comment]
replaced by #207 to separating upgrade circle ci code from gulp update.

**Pull request closed without merging**